src/dired.c

```diff
@@ -47,8 +47,35 @@ extern int is_slow_fs (const char *);
 #endif
 
 static ptrdiff_t scmp (const char *, const char *, ptrdiff_t);
-static Lisp_Object file_attributes (int, char const *, Lisp_Object,
-				    Lisp_Object, Lisp_Object);
+static Lisp_Object file_attributes_static (int, char const *, Lisp_Object,
+					   Lisp_Object, Lisp_Object);
+
+#ifndef WINDOWSNT
+extern Lisp_Object file_attributes(Lisp_Object, Lisp_Object);
+
+Lisp_Object file_attributes_c_internal (char const *,
+					Lisp_Object,
+					Lisp_Object,
+					Lisp_Object);
+Lisp_Object filemode_string(Lisp_Object);
+#endif
+
+extern Lisp_Object file_attributes_rust_internal (Lisp_Object,
+						  Lisp_Object,
+						  Lisp_Object);
+#ifdef WINDOWSNT
+Lisp_Object file_attributes_c(Lisp_Object, Lisp_Object);
+
+/*
+ * Workaround for
+ * https://github.com/Wilfred/remacs/issues/804
+ */
+static int
+dummy_file_attributes_rust_internal(void) {
```

Comment:
I don't see why you need this function.

Reply:
Given the windows build and https://github.com/Wilfred/remacs/issues/804 ...

```file_attributes_rust_internal``` is exported  (it really does not need to be for windows but #804).

I believe -- forget now if I ran into this or guessed I would -- since ```file_attributes_rust_internal``` was exported from rust dired it needed to be called at least once from another module else build would complain/fail.

And the dummy_ func wrapper is just a way to make it more clear it's not doing anything useful.

Reply:
What error do you get if you remove `dummy_file_attributes_rust_internal`?

Reply:
Was this related to the build.rs not parsing #cfg correctly? So it may not be needed.

Reply:
@db48x here's an example of the type of error
```
#[no_mangle]
pub extern "C" fn file_attributes_foo() -> bool {
    true
}
```
which gives
```
Compiling remacs v0.1.0 (file:///home/gb/warez/remacs-final/remacs/rust_src)

 warning: function is never used: `file_attributes_foo`
   --> src/dired.rs:78:1
    |
 78 | pub extern "C" fn file_attributes_foo() -> bool {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(dead_code)] on by default
```
This was a quick example on Linux. Given the state of the windows build it's hard to repro the real thing.

Reply:
@shaleh yes it's ```cfg``` not able to restrict an exported fn (#804) to just one of unix/windows. What not needed? (the dummy_ fn for windows?, support for windows anymore in current dev projects?)

Reply:
After the switch to importing unix/windows as modules I saw an easy way to remove the #804 workaround -- including the dummy_ C func -- next push will have it removed.